File: adjacencymatrix.py
# Write python program that accepts the vertices and edges for a graph and stores
# it as an adjacency matrix.

def create_adjacency_matrix(vertices, edges):
    # Initialize adjacency matrix with zeros
    adjacency_matrix = [[0] * len(vertices) for _ in range(len(vertices))]
    
    # Fill in the adjacency matrix based on the edges
    for edge in edges:
        start_vertex, end_vertex = edge
        start_index = vertices.index(start_vertex)
        end_index = vertices.index(end_vertex)
        adjacency_matrix[start_index][end_index] = 1
        adjacency_matrix[end_index][start_index] = 1  # For undirected graph
        
    return adjacency_matrix

# Example usage:
vertices = ['A', 'B', 'C', 'D']
edges = [('A', 'B'), ('B', 'C'), ('C', 'D'), ('D', 'A')]
adjacency_matrix = create_adjacency_matrix(vertices, edges)

# Print the adjacency matrix
print("Adjacency Matrix:")
for row in adjacency_matrix:
    print(row)





# This program defines a function create_adjacency_matrix() that takes a list of vertices and a list of edges as input and returns the corresponding adjacency matrix. It initializes an empty matrix with zeros and iterates through the edges to set the appropriate entries in the adjacency matrix. For an undirected graph, it sets both the source-to-destination and destination-to-source entries to 1.

File: adjacency_matrix.c
#include <stdio.h>

#define MAX_VERTICES 20

int main() {
    int vertices, edges;
    int graph[MAX_VERTICES][MAX_VERTICES] = {0};

    printf("Enter the number of vertices: ");
    scanf("%d", &vertices);

    printf("Enter the number of edges: ");
    scanf("%d", &edges);

    printf("Enter the edges (format: vertex1 vertex2):\n");
    for (int i = 0; i < edges; ++i) {
        int v1, v2;
        scanf("%d %d", &v1, &v2);
        if (v1 >= 0 && v1 < vertices && v2 >= 0 && v2 < vertices) {
            graph[v1][v2] = 1;
            graph[v2][v1] = 1; // assuming undirected graph
        } else {
            printf("Invalid edge: %d %d\n", v1, v2);
            --i; // decrement i to re-enter this edge
        }
    }

    printf("Adjacency Matrix:\n");
    for (int i = 0; i < vertices; ++i) {
        for (int j = 0; j < vertices; ++j) {
            printf("%d ", graph[i][j]);
        }
        printf("\n");
    }

    return 0;
}


//Test cases
// Enter the number of vertices: 5
// Enter the number of edges: 7
// Enter the edges (format: vertex1 vertex2):
// 0 1
// 0 2
// 1 2
// 1 3
// 2 3
// 2 4
// 3 4

// Adjacency Matrix:
// 0 1 1 0 0 
// 1 0 1 1 0 
// 1 1 0 1 1 
// 0 1 1 0 1 
// 0 0 1 1 0 
